### PR TITLE
WEBDOM-391: Convert build/destroy event listeners to tagged services.

### PR DESCRIPTION
**Pull request by Jelle-S**
WARNING: BC break!

### CODE REVIEW

**Review by daften — CHANGES_REQUESTED** [review]
Update to use e.g isProcessed or to return the task status when it has been run. Or at least evaluate

File: Service/TaskService.php [hunk 1]
```diff
@@ -32,66 +28,62 @@ class TaskService
...
-
-            $task->setStatus($status);
+        if ($task->isInProgress()) {
+            $task->setProcessed();
         }
 
         // Add a log message or simply persist any changes.
-        switch ($task->getStatus()) {
-            case Task::STATUS_PROCESSED:
+        switch (true) {
```
Comment (matthijs-va) [inline Service/TaskService.php]: This feels like a dirt trick.

File: Entity/Task.php [hunk 2]
```diff
@@ -146,4 +146,86 @@ public function setLog($log)
 
         return $this;
     }
+
```
Comment (matthijs-va) [inline Entity/Task.php:149]: I'n not a big fan of a method for each status.

Comment (Jelle-S) [reply Entity/Task.php:149]: Better readability IMHO

File: Service/TaskService.php (hunk 1 above)
Comment (Jelle-S) [reply Service/TaskService.php]: It's a pretty standard PHP feature, no?

File: Service/TaskRunnerService.php [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        try {
+            foreach ($provisioners as $provisioner) {
+                $provisioner->run($task);
+                if ($task->isFailed()) {
+                    break;
+                }
+            }
+        } catch (\Exception $ex) {
+            if (!($ex instanceof LoggedException)) {
+                $task->setFailed();
```
Comment (matthijs-va) [inline Service/TaskRunnerService.php]: I would move this one out of the if().

File: Service/TaskRunnerService.php [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                $this->logger->addErrorLogMessage($task, $ex->getMessage(), 2);
```
Comment (matthijs-va) [inline Service/TaskRunnerService.php]: Suggestion: use method chaining.

**Review by matthijs-va — CHANGES_REQUESTED** [review]

**Review by matthijs-va — APPROVED** [review]

**Pull request merged**